src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/ClientServices.cs

```diff
@@ -258,46 +444,56 @@ public class ClientServices : IClientHost, IEndpointServices, IEndpointDiscovery
         /// <returns></returns>
         internal IClientSession GetOrCreateSession(EndpointIdentifier id, bool persistent) {
             return _clients.GetOrAdd(id, k => new ClientSession(
-                CreateApplicationConfiguration(TimeSpan.FromMinutes(2), TimeSpan.FromMinutes(2)),
-                k.Endpoint, () => Certificate, _logger, NotifyStateChangeAsync, persistent,
+                _OpcApplicationConfig, k.Endpoint, () => Certificate, _logger, NotifyStateChangeAsync, persistent,
                     _maxOpTimeout));
         }
 
         /// <summary>
         /// Create application configuration for client
         /// </summary>
         /// <returns></returns>
-        internal static ApplicationConfiguration CreateApplicationConfiguration(
+        internal ApplicationConfiguration CreateApplicationConfiguration(
             TimeSpan operationTimeout, TimeSpan sessionTimeout) {
+
             return new ApplicationConfiguration {
                 ApplicationName = "Azure IIoT OPC Twin Client Services",
                 ApplicationType = Opc.Ua.ApplicationType.Client,
-                ApplicationUri =
-            "urn:" + Utils.GetHostName() + ":OPCFoundation:CoreSampleClient",
+                ApplicationUri ="urn:" + Utils.GetHostName() + ":Azure:IIoTOpcTwin",
+                CertificateValidator = new Opc.Ua.CertificateValidator(),
                 SecurityConfiguration = new SecurityConfiguration {
                     ApplicationCertificate = new CertificateIdentifier {
                         StoreType = "Directory",
-                        StorePath =
-            "OPC Foundation/CertificateStores/MachineDefault",
-                        SubjectName = "UA Core Sample Client"
+                        StorePath = _configuration.ApplicationCertificateFolder ??
+                                    (_configuration.PkiRootFolder != null ? 
+                                        _configuration.PkiRootFolder + "/pki/own" :
```

Comment:
Werent those subfolders also in the config?

Reply:
no, they were not, but they are now.